### PR TITLE
Add an rtr workflow to the vulnerability scanner module - Implementation

### PR DESCRIPTION
**Pull request by sebasfalcone**
|Related issue|
|---|
| #17952 |

## Description
This PR aims to add the baseline for the vulnerability CI testing on GitHub actions.

It includes base testing, which is compiling the different modules and executing the associated tests

The CMake files were changed to allow the compilation of the individual modules separately

### CODE REVIEW

File: .github/workflows/vulnerability-scanner-tests.yml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          fetch-depth: 2 # We need to fetch 2 commits to get the diff.
+
+      # Checks which modules has changed
```
Comment (tsarquis88) [inline .github/workflows/vulnerability-scanner-tests.yml:28]: Typo
```suggestion
      # Checks which modules have changed
```

File: .github/actions/compile_and_test/action.yml
Comment (tsarquis88) [inline .github/actions/compile_and_test/action.yml:1]: Good job here, very well re-used.

File: .github/workflows/vulnerability-scanner-tests.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    types: [synchronize, opened, reopened, ready_for_review]
+    
+    # Path filtering
+    paths:
+      - ".github/workflows/vulnerability-scanner-tests.yml"
+      - "src/shared_modules/router/**"
+      - "src/shared_modules/content_manager/**"
+      - "src/shared_modules/indexer_connector/**"
+      - "src/wazuh_modules/vulnerability_scanner/**"
+      - "src/wazuh_modules/wazuh-http-request/**"
```
Comment (tsarquis88) [inline .github/workflows/vulnerability-scanner-tests.yml:15]: Is this unused?

File: .github/workflows/vulnerability-scanner-tests.yml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+jobs:
+  vulnerability-scanner-modules:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          fetch-depth: 2 # We need to fetch 2 commits to get the diff.
```
Comment (tsarquis88) [inline .github/workflows/vulnerability-scanner-tests.yml:26]: What would happen if I push more than two commits?
```
...
git commit -m "Commit a"
...
git commit -m "Commit b"
...
git commit -m "Commit c"
git push
```

Comment (sebasfalcone) [reply .github/workflows/vulnerability-scanner-tests.yml:26]: Good question.. maybe I should check the diffs against the base head?

File: .github/workflows/vulnerability-scanner-tests.yml (hunk 2 above)
Comment (sebasfalcone) [reply .github/workflows/vulnerability-scanner-tests.yml:15]: It prevents the workflow from running when files other than the ones in te list are modified

Comment (tsarquis88) [reply .github/workflows/vulnerability-scanner-tests.yml:15]: Aaah okay, is a built-in [feature](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-files-changed-in-a-pull-request). Nice.

File: .github/workflows/vulnerability-scanner-tests.yml (hunk 3 above)
Comment (tsarquis88) [reply .github/workflows/vulnerability-scanner-tests.yml:26]: Not quite sure. If the trigger is a pull request, things get complicated. 
Maybe using [GitHub CLI](https://cli.github.com/manual/gh_pr)? [Here](https://github.com/godotengine/godot/blob/master/.github/workflows/static_checks.yml#L34) is an example of how that could be achieved.

Comment (sebasfalcone) [reply .github/workflows/vulnerability-scanner-tests.yml:26]: Found a generic solution:
`git diff --name-only origin/${{ github.head_ref } origin/${{ github.base_ref }})`

github.head_ref -> The HEAD of the current branch
github.base_ref -> The HEAD of the branch where the PR points

Comment (sebasfalcone) [reply .github/workflows/vulnerability-scanner-tests.yml]: Seems like I was wrong. Also, to apply your suggestion we need to have a token from the repo admins :/

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**